src/pgwire/src/codec.rs

```diff
@@ -36,6 +38,7 @@ use crate::message::{BackendMessage, BackendMessageKind};
 /// A connection that manages the encoding and decoding of pgwire frames.
 pub struct FramedConn<A> {
     conn_id: ConnectionId,
+    pub peer_addr: Option<IpAddr>,
```

Comment:
For consistency with `conn_id`, we should keep this private and add a public accessor method. Doesn't matter too much, but nice for the API to enforce that the client address can't change after the connection is created.